src/components/base/base-search-box.tsx

```diff
@@ -102,16 +115,26 @@ export const BaseSearchBox = (props: SearchProps) => {
     };
   }, [matchCase, matchWholeWord, useRegularExpression, t]);
 
-  useEffect(() => {
+  const updateSearch = useCallback(() => {
```

Comment:
`updateSearch` is only lived in `useEffect`, there is no need to hoist it and wrap it in `useCallback`.

Also TBH `updateSearch` should live in the input's `onChange` callback, not in `useEffect`.